frontend/src/pages/PointsOfInterestView.vue

```diff
@@ -1,132 +1,126 @@
+
+const pois = ref<POIDTO[]>([]);
+const cities = ref<CityDTO[]>([]);
+const selectedPOIs = ref<number[]>([])
+const currentPOI = ref<number>();
+
+const initialValues = ref<POIDTO>({
+    name: '',
+    description: '',
+    cityId: 0,
+    price: 0,
+    type: ''
+});
+
+const types = ref<string[]>([]);
+const formKey = ref<number>(0);
+
+await fetchPOIs();
+await fetchCities();
+await fetchTypes();
+
+async function fetchPOIs() {
+    pois.value = await poiApi.getPointsOfInterest();
+}
+
+async function fetchCities() {
+    cities.value = await cityApi.getCities();
+}
+
+async function fetchTypes() {
+    types.value = await poiApi.getAllPointsOfInterestTypes();
+}
+
+function select(poi: POIDTO) {
+    const poiAlreadySelected = selectedPOIs.value.includes(poi.id!);
+    if (poiAlreadySelected) {
+        selectedPOIs.value = selectedPOIs.value.filter((poiId) => poiId !== poi.id);
+        return;
     }
+    selectedPOIs.value.push(poi.id!);
+}
 
-    async function fetchCities() {
-        cities.value = await cityApi.getCities();
+async function deletePOI() {
+    for (const id of selectedPOIs.value) {
+        await poiApi.deletePointOfInterest({id});
     }
-
-    async function fetchTypes() {
-        types.value = await poiApi.getAllPointsOfInterestTypes();
+    showToast('success', t('poi.deleted'), toast);
+    selectedPOIs.value = [];
+    await fetchPOIs()
+    inEditingMode.value = EditMode.NONE;
+    initialValues.value = {};
+    formKey.value++;
+}
+
+function onCardClick(poi: POIDTO) {
+    inEditingMode.value = EditMode.UPDATE;
+    initialValues.value = poi;
+    currentPOI.value = poi.id;
+    formKey.value++;
+}
+
+function onAddClick(){
+    inEditingMode.value = EditMode.ADD;
+    initialValues.value = {};
+    formKey.value++;
+}
+
+function resolver(submitEvent: FormResolverOptions) {
+    const errors: FormErrors = {};
+
+    if (submitEvent.values.name.trim().length < 2) {
+        errors.name = [{message: t('formFieldError.fieldLength.min', {lowerBound: 2})}];
     }
-
-    function select(poi: POIDTO) {
-        const poiAlreadySelected = selectedPOIs.value.includes(poi.id!);
-        if (poiAlreadySelected) {
-            selectedPOIs.value = selectedPOIs.value.filter((poiId) => poiId !== poi.id);
-            return;
+    for(const k in submitEvent.values){
+        const v = submitEvent.values[k];
+        if (v === "" || v === null || v === undefined){
+            errors[k] = [{message: t('formFieldError.fieldRequired')}];
         }
-        selectedPOIs.value.push(poi.id!);
     }
 
-    async function deletePOI() {
-        for (const id of selectedPOIs.value) {
-            await poiApi.deletePointOfInterest({id});
+    return {
+        values: submitEvent.values,
+        errors
+    };
+}
+
+async function onFormSubmit(submitEvent: FormSubmitEvent) {
+    if (submitEvent.valid) {
+        if(inEditingMode.value==EditMode.ADD) {
+            await poiApi.addPointOfInterest({pOIDTO: submitEvent.values});
```

Comment:
Extract these 3 lines to their own function like you've done with the deletePOI.
```suggestion
            async function sendPOICreationRequest(poi) {
                  await poiApi.addPointOfInterest({pOIDTO: poi});
                  showToast(...);
                  await fetchPOIs();
            }
```